src/game/object_helpers.c

```diff
@@ -839,6 +839,33 @@ struct Object *cur_obj_find_nearest_object_with_behavior(const BehaviorScript *b
     return closestObj;
 }
 
+struct Object *cur_obj_find_nearest_object_with_bparam_2nd_byte(const BehaviorScript *behavior, f32 *dist, u8 bparam) {
```

Comment:
maybe the name could be more specific, at first I thought that this didn't care about a matching behavior as much as it did the bparam. something like `cur_obj_find_nearest_object_with_bhv_and_bparam`

Reply:
This might be overly complex or difficult to convey easily, but what if you could pass both a u32 representing bparams and a bitmask to specify which bparams you actually care about? Because as it stands now, this is just the function above with 1 extra conditional check. I'm all for useful functions but I don't want to set a standard where we open the door to 700 near-duplicates.

Reply:
what about byte index? then the function can make the mask

Reply:
Byte index isn't good enough if you need to compare against more than one parameter or even just a subset of a parameter. Using a bitmask, if implemented properly, shouldn't have any runtime disadvantages either, but admittedly may make the function more confusing to use.